### PR TITLE
Add Chrysalis machine config files

### PR DESCRIPTION
**Pull request by amametjanov**
Add Chrysalis machine config files.
Change anvil paths to use new /lcrc/group/e3sm disk.   Remove "acme" from paths.
Change "acme_scratch" to just "scratch".

Update CIME submodule to hash  5cef5558c4e7bebfc8ec8f3b507e7e583e6a669c
  - Print TPUTCOMP difference to TestStatus.log
  - Update OS process id error-checking in GPTL's get_memusage

CIME changes that are only for CESM
- Changes to implement ESMF aware threading in the nuopc driver. Correct issues with nuopc timing file.
- Update modules on cheyenne.
- Changes for nuopc xcpl_comps needed for new ice sheets capability
- Make the default pio_version = 2

[NML] - on Anvil due to inputdata file pathname changes in seq_maps.rc namelist

### CODE REVIEW

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -1119,6 +1119,79 @@
...
+    <DESC>ANL LCRC cluster 512-node AMD Epyc 7532 2-sockets 64-cores per node</DESC>
+    <NODENAME_REGEX>chrlogin.*</NODENAME_REGEX>
+    <OS>LINUX</OS>
+    <COMPILERS>intel</COMPILERS>
+    <MPILIBS>openmpi</MPILIBS>
+    <PROJECT>e3sm</PROJECT>
+    <SAVE_TIMING_DIR>/lcrc/group/e3sm</SAVE_TIMING_DIR>
+    <SAVE_TIMING_DIR_PROJECTS>.*</SAVE_TIMING_DIR_PROJECTS>
+    <CIME_OUTPUT_ROOT>/lcrc/group/e3sm/$USER/scratch/chrys</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>/lcrc/group/e3sm/ccsm-data/inputdata</DIN_LOC_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:1132]: Make this /lcrc/group/e3sm/data/inputdata

**Comment by amametjanov** [comment]
Ran e3sm_integration + e3sm_prod test suites on Anvil as a sanity check. All tests pass except for 1 expected fail:
```
@blueslogin3:20201215-integr$ ./cs.status.20201215_081028_gwdn4p | grep "Overall: " | wc -l
91
@blueslogin3:20201215-integr$ ./cs.status.20201215_081028_gwdn4p | grep "Overall: PASS" | wc -l
90
@blueslogin3:20201215-integr$ ./cs.status.20201215_081028_gwdn4p | grep "Overall: FAIL"
  SMS_B.ne4_ne4.F-EAMv1-AQP1.anvil_intel.eam-hommexx (Overall: FAIL) details:
```

Jim (@jgfouca), please take a look at the CIME updates in this PR. Thanks.

**Comment by rljacob** [comment]
You should summarize any other changes to CIME coming in.

File: cime_config/machines/config_machines.xml [hunk 2]
```diff
@@ -1119,20 +1119,110 @@
...
-    <DIN_LOC_ROOT>/home/ccsm-data/inputdata</DIN_LOC_ROOT>
-    <DIN_LOC_ROOT_CLMFORC>/home/ccsm-data/inputdata/atm/datm7</DIN_LOC_ROOT_CLMFORC>
-    <DOUT_S_ROOT>/lcrc/group/acme/$USER/archive/$CASE</DOUT_S_ROOT>
-    <BASELINE_ROOT>/lcrc/group/acme/acme_baselines/blues/$COMPILER</BASELINE_ROOT>
-    <CCSM_CPRNC>/lcrc/group/acme/tools/cprnc/cprnc</CCSM_CPRNC>
+    <CIME_OUTPUT_ROOT>/lcrc/group/e3sm/$USER/scratch/blues</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>/lcrc/group/e3sm/data/inputdata</DIN_LOC_ROOT>
+    <DIN_LOC_ROOT_CLMFORC>/lcrc/group/e3sm/data/inputdata/atm/datm7</DIN_LOC_ROOT_CLMFORC>
+    <DOUT_S_ROOT>/lcrc/group/e3sm/$USER/archive/$CASE</DOUT_S_ROOT>
+    <BASELINE_ROOT>/lcrc/group/e3sm/baselines/blues/$COMPILER</BASELINE_ROOT>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:1224]: This might disrupt the testing (everything will diff).  @jgfouca ?

**Comment by rljacob** [comment]
Please add in the description the CIME hash that the submodule now points to.

**Comment by rljacob** [comment]
Did you also change paths on bebop?

File: cime_config/machines/config_machines.xml (hunk 2 above)
Comment (amametjanov) [reply cime_config/machines/config_machines.xml:1224]: Yes. Baselines for nightly testing are kept just for anvil+e3sm_prod (3 tests). Existing baselines can be copied over or new baselines generated.

**Comment by amametjanov** [comment]
Added new CIME hash. Bebop paths are also updated (anvil, blues, bebop).

**Review by rljacob — APPROVED** [review]

File: cime_config/machines/config_machines.xml (hunk 2 above)
Comment (jgfouca) [reply cime_config/machines/config_machines.xml:1224]: @amametjanov , lets copy them. That way, we can maintain bless and runtime metrics.

**Review by jgfouca — APPROVED** [review]
The CIME changes look fine.

File: cime_config/machines/config_machines.xml (hunk 2 above)
Comment (amametjanov) [reply cime_config/machines/config_machines.xml:1224]: Done (~830 cases with walltimes).
`e3sm_prod` is BFB: copied baselines vs. this branch rebased locally on top of Dec 16 master (hash 2bc0cc0).
Merging to `next` for a round of testing of cime changes on other machines.

**Pull request merged**